src/libs/store/createStore.ts

```diff
@@ -0,0 +1,40 @@
+import AsyncStorage from '@react-native-async-storage/async-storage'
+import { create, StateCreator } from 'zustand'
+import { createJSONStorage, devtools, persist } from 'zustand/middleware'
+
+type Options = {
+  persist?: boolean
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any -- we don't want to restrict the payload type
+export const createStore = <State, Actions extends Record<string, (payload: any) => void>>(
```

Comment:
je recommande `unknown` plutot que `any`

ça permet d'avoir plus de garantie de TS

```suggestion
export const createStore = <State, Actions extends Record<string, (payload: unknown) => void>>(
```

Reply:
J'avais essayé, mais ça casse le typage à la création du store : il s'attend à ce que les actions prennent unknown en entrée, alors que non, elles peuvent réellement prendre n'importe quoi en entrée

Reply:
dommage